test/test_helper.rb

```diff
@@ -178,6 +178,23 @@ def force_publish(edition)
   def fixture_path
     Pathname.new(Rails.root.join("test/fixtures"))
   end
+
+  def setup_fog_mock
+    Fog.mock!
+    ENV["AWS_REGION"] = "eu-west-1"
+    ENV["AWS_ACCESS_KEY_ID"] = "test"
+    ENV["AWS_SECRET_ACCESS_KEY"] = "test"
+    ENV["AWS_S3_BUCKET_NAME"] = "test-bucket"
+
+    # Create an S3 bucket so the code being tested can find it
+    connection = Fog::Storage.new(
+      provider: "AWS",
+      region: ENV["AWS_REGION"],
+      aws_access_key_id: ENV["AWS_ACCESS_KEY_ID"],
+      aws_secret_access_key: ENV["AWS_SECRET_ACCESS_KEY"],
+    )
+    @directory = connection.directories.get(ENV["AWS_S3_BUCKET_NAME"]) || connection.directories.create(key: ENV["AWS_S3_BUCKET_NAME"]) # rubocop:disable Rails/SaveBang
```

Comment:
Could this be simplified to this?

```ruby
    @directory ||= connection.directories.create(key: ENV["AWS_S3_BUCKET_NAME"])
```

Reply:
Looks like this might be some over-cautious coding, since in production the bucked will always exist and in tests it will always have to be created. `||=` won't work because it will be `nil` initially either way, but the `get` can be dropped, I think.